### ISSUE TITLE
Nothing running on port 50001 on odroid

### ISSUE THREAD

**Issue by justinmoon** [romanz/electrs#156]
I'm trying to run this on an Odroid HC2 according to [this guide](https://github.com/openoms/bitcoin-tutorials/tree/master/electrs):

The prometheus service is running:

```
$ lsof -i :4224
COMMAND   PID  USER   FD   TYPE  DEVICE SIZE/OFF NODE NAME
electrs 14224 admin    5u  IPv4 4035845      0t0  TCP localhost:xtell (LISTEN)
```

But the electrum service doesn't seem to be running

```
$ lsof -i :50001
(nothing)
```

I ran it for a day and the data directory filled up so that part is working:

```
$ sudo du -h /mnt/hdd/electrs/db/
50G	/mnt/hdd/electrs/db/mainnet
50G	/mnt/hdd/electrs/db/
```

When I run with the  `-v` flag I don't really get any obvious errors:

```
Config { log: StdErrLog { verbosity: Warn, quiet: false, timestamp: Off, modules: [], writer: "stderr", color_choice: Auto }, network_type: bitcoin, db_path: "/mnt/hdd/electrs/db/mainnet", daemon_dir: "/home/admin/.bitcoin", daemon_rpc_addr: V4(127.0.0.1:8332), cookie: Some("raspibolt:<my-rpc-password>"), electrum_rpc_addr: V4(0.0.0.0:50001), monitoring_addr: V4(127.0.0.1:4224), jsonrpc_import: true, index_batch_size: 10, bulk_index_threads: 8, tx_cache_size: 10000, txid_limit: 100, server_banner: "Welcome to electrs 0.7.0 (Electrum Rust Server)!", blocktxids_cache_size: 100 }
```

**Comment by justinmoon** [comment]
Ahh when i ran with `-vvvv` I get more information:

```
DEBUG - Server listening on 127.0.0.1:4224
DEBUG - Running accept thread
INFO - NetworkInfo { version: 170100, subversion: "/Satoshi:0.17.1/" }
INFO - BlockchainInfo { chain: "main", blocks: 585708, headers: 585708, bestblockhash: "00000000000000000012ac3ad31bcdd9f69dea68f5629c6b9e358c7e8ce1cc25", pruned: false, initialblockdownload: false }
DEBUG - opening DB at "/mnt/hdd/electrs/db/mainnet"
TRACE - lastest indexed blockhash: 00000000000000000007762c7f4a91fd7d0c4ccae65f36f0b0dbd68965133f35
DEBUG - applying 543930 new headers from height 0
DEBUG - downloading new block headers (543930 already indexed) from 00000000000000000012ac3ad31bcdd9f69dea68f5629c6b9e358c7e8ce1cc25
TRACE - downloaded 41779 block headers
INFO - best=00000000000000000012ac3ad31bcdd9f69dea68f5629c6b9e358c7e8ce1cc25 height=585708 @ 2019-07-16T19:01:57Z (41779 left to index)
```

Seems I have a few headers left to process. Was told data dir would be about 50 gb so erroneously assumed it was done. If this is the problem, it is still odd that the electrum server wouldn't start until all headers have been processed. IIRC the python Electrum Personal Server will allow you to connect before indexing completes and you'd get a yellow connection icon in Electrum desktop.

**Comment by romanz** [comment]
>  it is still odd that the electrum server wouldn't start until all headers have been processed.

#543 should help - to be released soon @ #628 :)
